packages/expo-cli/src/commands/upload.ts

```diff
@@ -134,18 +135,11 @@ export default function (program: Command) {
     });
 }
 
-function checkRuntimePlatform(targetPlatform: 'android' | 'ios'): void {
+function assertRuntimePlatform(targetPlatform: 'android' | 'ios'): void {
   if (process.platform !== 'darwin') {
     if (targetPlatform === 'android') {
-      log.error('Local Android uploads are only supported on macOS.');
-      log(
-        chalk.bold(
-          'Try the --use-submission-service flag to upload your app from Expo servers. This feature is behind a flag because it is experimental.'
-        )
-      );
-    } else {
-      log.error('Currently, iOS uploads are only supported on macOS, sorry :(');
+      throw new CommandError('Local Android uploads are only supported on macOS.');
     }
```

Comment:
Correct me if I'm missing sth but we don't need this `if` at all. This function is only called for `upload:ios` 🤔

Reply:
Yep, the android part is never called anymore, we don't have `Local Android uploads`